recipes/nmos-cpp/all/conanfile.py

```diff
@@ -40,12 +40,20 @@ def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
+        # Simple change as github doesn't allow review comments on unchanged lines that are outside the +/- 3 lines of the changes.
+        # To be removed in the review process.
         if self.settings.os == "Macos":
             del self.options.with_dnssd
+        # The removal on macos makes sense as the mdnsresponder (which is by apple) comes with the OS/Xcode SDK I guess,
+        # but why override the default. Took me (luckily only a few minutes) time to see why it wanted to get avahi despite the default_options and like thousands of
+        # dependencies. mdnsresponder is just itself, avahi really has a large sets of dependencies and many of them LGPL infested...
+        # Also getting the `conan graph info ... --format=html` for avahi and mdnsresponder and opening both in the browser is eye-opening. If there is a reason
+        # why mdnsresponder isn't cutting it, would be nice to be noted. I'd prefer it with the two elif cases removed to have some consistency.
         elif self.settings.os == "Linux":
             self.options.with_dnssd = "avahi"
         elif self.settings.os == "Windows":
             self.options.with_dnssd = "mdnsresponder"
+        # a last one to being able to github this ...
 
```

Comment:
```suggestion
            del self.options.with_dnssd

```
So I wonder why not to remove this or what the reason is for that.

Maybe @garethsb could give some input why there is an override of the defaults. If there is a good reason, I think it should be documented (and I'd happily add that, here, where the override happens and with a short comment in the `default_options` so that others don't need to wonder why despite the default avahi is pulled).

Otherwise maybe @jonathan-r-thorpe or @lo-simon who seem to be main contributors to nmos-cpp can comment?